run.py

```diff
@@ -39,14 +44,14 @@ def get_args():
       sys.exit("Implementation " + name + " not found.")
     implementations[name] = image
 
-def get_impls(arg) -> dict:
+def get_impls(arg, availableImpls, role) -> List[str]:
   if not arg:
-    return implementations
-  impls = {}
+    return availableImpls
+  impls = []
   for s in arg.split(","):
-    if s not in implementations:
-      sys.exit("Implementation " + s + " not found.")
-    impls[s] = implementations[s]
+    if s not in availableImpls:
```

Comment:
I would suggest leaving this as it is, and simply skipping the impl in run()

Reply:
Sorry, i don't get it. If you review my change in the side-by-side view, you'll see that i kept the logic (about skipping or not skipping an implementation) identical to what it was before. The main thing i changed is that the implementations are not taken from the global "implemenations" variable, but from a function parameter, so it can be (re)used to either get_impls for client implementations or for server implementations.